src/components/recordedit/form-row.tsx

```diff
@@ -0,0 +1,285 @@
+// components
+import InputSwitch from '@isrd-isi-edu/chaise/src/components/input-switch/input-switch';
+import SelectAllRow from '@isrd-isi-edu/chaise/src/components/recordedit/select-all-row';
+
+// hooks
+import { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import useRecordedit from '@isrd-isi-edu/chaise/src/hooks/recordedit';
+
+// models
+import { appModes, SELECT_ALL_INPUT_FORM_VALUE } from '@isrd-isi-edu/chaise/src/models/recordedit';
+
+// utils
+import { getDisabledInputValue } from '@isrd-isi-edu/chaise/src/utils/input-utils';
+import ResizeSensor from 'css-element-queries/src/ResizeSensor';
+import { isObjectAndKeyDefined } from '@isrd-isi-edu/chaise/src/utils/type-utils';
+import { makeSafeIdAttr } from '@isrd-isi-edu/chaise/src/utils/string-utils';
+
+type FormRowProps = {
+    columnModelIndex: number;
+    removeFormIndex?: number;
+    removeClicked?: boolean;
+    setRemoveClicked?: any;
+};
+const FormRow = ({
+    columnModelIndex,
+    removeFormIndex,
+    removeClicked,
+    setRemoveClicked,
+}: FormRowProps): JSX.Element => {
+    const {
+        forms,
+        appMode,
+        reference,
+        columnModels,
+        tuples,
+        activeSelectAll,
+        canUpdateValues,
+        columnPermissionErrors,
+        foreignKeyData,
+        waitingForForeignKeyData,
+        getRecordeditLogStack,
+        getRecordeditLogAction,
+    } = useRecordedit();
+
+    // This state variable is to set the form as active when its selected
```

Comment:
In your comment mention whether you're storing the `formIndex` or `formNumber`. I'm assuming it's `formNumber`